mcp/server.go

```diff
@@ -224,6 +236,12 @@ func (s *Server) capabilities() *serverCapabilities {
 	if s.resources.len() > 0 || s.resourceTemplates.len() > 0 {
 		caps.Resources = &resourceCapabilities{ListChanged: true}
 	}
+	if s.opts.SubscribeHandler != nil {
```

Comment:
Hmm.
I think that if there are no resources, it doesn't matter if there's a subscribe handler. There is no resource capability.
It's OK to provide a handler, it just doesn't do anything.